test/frameworks/karma.test.js

```diff
@@ -0,0 +1,229 @@
+
+const { Karma } = require('../../src/frameworks/karma');
+const { Helpers } = require('../../src/helpers');
+
+const { expect } = require('chai');
+const sinon = require('sinon');
+
+describe('Karma', () => {
+
+	describe('updateTestCommand', () => {
+		it('Returns same command', () => {
+			const command = 'ls';
+			const fileOutput = 'output.json';
+
+			expect(Karma.updateTestCommand(command, fileOutput))
+				.to.equal(`/bin/bash -c "${command} | tee ${fileOutput}"`);
+		});
+
+		it('Returns modified command', () => {
+			const fileOutput = 'output.json';
+
+			expect(Karma.updateTestCommand('karma', fileOutput))
+				.to.equal(`/bin/bash -c "karma --reporters json | tee ${fileOutput}"`);
+		});
+	});
+
+	describe('parseTestResult', () => {
+		const ANNOTATION_TITLE = 'Some unit test(s) have failed';
+		const ANNOTATION_SUMMARY = 'failure(s) were found';
+
+		beforeEach(() => {
+			sinon.stub(Helpers, 'readFile');
+		});
+
+		afterEach(() => {
+			sinon.restore();
+		});
+
+		it('Parses no errors', () => {
+			sinon.stub(Helpers, 'extractJSON').returns({
+				summary: {
+					failed: 0
+				}
+			});
+
+			const result = Karma.parseTestResult('');
+
+			expect(result.title).to.equal(ANNOTATION_TITLE);
+			expect(result.summary).to.equal(`0 ${ANNOTATION_SUMMARY}`);
+			expect(result.annotations.length).to.equal(0);
+		});
+
+		it('Parses 1 error', () => {
+			const line = 42;
+			const filePath = 'test/somewhere/here.test.js';
+			const title = 'error title';
+			const error_message = 'error message';
+			const error_stack = 'ImaginaryException: Error happened';
+			const browser_name = 'Chrome';
+			const suite = [
+				'functionTested',
+				'Use Case'
+			];
+
+			sinon.stub(Karma, '_getFileInfo').returns({
+				path: filePath,
+				line
+			});
+
+			sinon.stub(Helpers, 'extractJSON').returns({
+				summary: {
+					failed: 1
+				},
+				browsers: {
+					123: {
+						name: browser_name
+					}
+				},
+				result: {
+					123: [
+						{
+							success: true
+						}, {
+							success: false,
+							log: [error_message, error_stack],
+							suite,
+							description: title
+						}, {
+							success: true
+						}
+					]
+				}
+			});
+
+			const result = Karma.parseTestResult('');
+
+			expect(result.title).to.equal(ANNOTATION_TITLE);
+			expect(result.summary).to.equal(`1 ${ANNOTATION_SUMMARY}`);
+			expect(result.annotations.length).to.equal(1);
+
+			expect(result.annotations[0].path).to.equal(filePath);
+			expect(result.annotations[0].start_line).to.equal(line);
+			expect(result.annotations[0].end_line).to.equal(line);
+			expect(result.annotations[0].annotation_level).to.equal('failure');
+			expect(result.annotations[0].message).to.equal(`${suite.join('.')}: ${title} (${browser_name}):\n\n${error_message}\n${error_stack}`);
+			expect(result.annotations[0].title).to.equal(`${title} (${browser_name})`);
+		});
+
+		it('Parses multiple errors', () => {
+			const browser_name = 'Chrome';
+
+			const line1 = 42;

```

Comment:
Nit: You could write a function to build you an `ErrorDetails` obj or something and it could "procedurally" generate some of this stuff, and then similarly you could write a separate function to assert the annotation based on this `ErrorDetails` obj